core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1309,4 +1309,11 @@ package object config {
     .bytesConf(ByteUnit.BYTE)
     .createOptional
 
+  private[spark] val EAGER_RDD_PARTITION_EVALUATE =
+    ConfigBuilder("spark.rdd.eager.partition")
```

Comment:
I haven't looked in detail, but I'm really skeptical of this. How will a user find this? if it's set one way, we still have the problem? if it's set another way, that changes a lot of behavior, potentially.

Reply:
Thanks @srowen I did some testing and these seems to solve the problem but i was skeptic of the change so kept it configurable so that user(admin) can change back if this introduces unwanted behavior. I did not intend to make this configuration public. DAGScheduler is still a grey area for me, would like to know any feedback for this change. If it checks all the boxes, i am too +1 for not making this configurable.

Reply:
Dropped the configuration